lib/rule_engine/utils.py

```diff
@@ -0,0 +1,38 @@
+import datetime
+import re
+
+
+_sub_regex = r'[0-9]+([,.][0-9]+)?'
+timedelta_regex = (
+    r'P(?!\b)'
+    f'(?P<weeks>{_sub_regex}W)?'
+    f'(?P<days>{_sub_regex}D)?'
+    r'(T'
+    f'(?P<hours>{_sub_regex}H)?'
+    f'(?P<minutes>{_sub_regex}M)?'
+    f'(?P<seconds>{_sub_regex}S)?'
+    r')?'
+)
+timedelta_re = re.compile(f"^{timedelta_regex}$")
+
+def parse_timedelta(periodstring):
+    if periodstring == "P":
+        raise ValueError('empty timedelta string')
+
+    match = timedelta_re.match(periodstring)
+    if not match:
+        raise ValueError('invalid timedelta string')
+
+    groups = match.groupdict()
+    for key, val in groups.items():
+        if val is None:
+            val = "0n"
+        groups[key] = float(val[:-1].replace(',', '.'))
```

Comment:
Is this replacing commas with periods for localization? I would have expected it to just remove command so `1,300` would become `1300` not `1.3`.

```
In [14]: parse_timedelta('P1,300D')
Out[14]: datetime.timedelta(days=1, seconds=25920)
```

Reply:
I'm still confused on this point. It seems like it might be for localization and if that's case, I'd really like for it to be configurable via the context or some other means, defaulting to en-US.

Reply:
Nevermind, I should have RTFM'ed. This behavior is actually called out in the standard or at least what's on Wikipedia because I guess you have to pay for ISO standards.

> The smallest value used may also have a decimal fraction,[[35]](https://en.wikipedia.org/wiki/ISO_8601#cite_note-35) as in "P0.5Y" to indicate half a year. This decimal fraction may be specified with either a [comma](https://en.wikipedia.org/wiki/Comma) or a [full stop](https://en.wikipedia.org/wiki/Full_stop), as in "P0,5Y" or "P0.5Y".

Source: https://en.wikipedia.org/wiki/ISO_8601#Durations

Reply:
👍